frontend/src/app/projects/models/model-source/t4c/create-t4c-model-new-repository/create-t4c-model-new-repository.component.ts

```diff
@@ -0,0 +1,132 @@
+import { T4CRepoService } from 'src/app/settings/modelsources/t4c-settings/service/t4c-repos/t4c-repo.service';
+import { ModelService } from '../../../service/model.service';
+import { T4CModelService } from '../service/t4c-model.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-create-t4c-model-new-repository',
+  templateUrl: './create-t4c-model-new-repository.component.html',
+  styleUrls: ['./create-t4c-model-new-repository.component.css'],
+})
+export class CreateT4cModelNewRepositoryComponent implements OnInit {
+  @Input() asStepper?: boolean;
+  @Output() create = new EventEmitter<boolean>();
+
+  private projectSlug?: string;
+  private modelSlug?: string;
+
+  constructor(
+    public projectService: ProjectService,
+    public modelService: ModelService,
+    public t4cModelService: T4CModelService,
+    public t4cInstanceService: T4CInstanceService,
+    private t4cRepoService: T4CRepoService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private toastService: ToastService
+  ) {}
+
+  public form = new FormGroup({
+    t4cInstanceId: new FormControl<number | null>(null, Validators.required),
+    t4cRepositoryName: new FormControl<string | null>(
+      { value: null, disabled: true },
+      Validators.required,
+      this.t4cRepoService.asyncNameValidator()
+    ),
+    t4cProjectName: new FormControl<string | null>(
+      { value: null, disabled: true },
+      Validators.required
+    ),
+  });
+
+  get t4cRepositoryNameControl() {
+    return this.form.controls.t4cRepositoryName;
+  }
+
+  get t4cProjectNameControl() {
+    return this.form.controls.t4cProjectName;
+  }
+
+  ngOnInit(): void {
+    this.t4cInstanceService.loadInstances();
+
+    combineLatest([
+      this.projectService.project$.pipe(filter(Boolean)),
+      this.modelService.model$.pipe(filter(Boolean)),
+    ])
+      .pipe(untilDestroyed(this))
+      .subscribe(([project, model]) => {
+        this.projectSlug = project.slug;
+        this.modelSlug = model.slug;
+      });
+
+    this.form.controls.t4cInstanceId.valueChanges.subscribe(
+      (selectedInstanceId) => {
+        if (selectedInstanceId) {
+          this.t4cRepoService.loadRepositories(selectedInstanceId);
+
+          this.t4cRepositoryNameControl.setValue(null);
+          this.t4cRepositoryNameControl.enable();
+
+          this.t4cProjectNameControl.setValue(null);
+          this.t4cProjectNameControl.enable();
+        }
+      }
+    );
+
+    this.t4cRepositoryNameControl.valueChanges.subscribe((value) => {
+      this.t4cProjectNameControl.setValue(value);
+    });
+  }
+
+  onSubmit(): void {
+    const projectSlug = this.projectSlug;
+    const modelSlug = this.modelSlug;
+
+    if (this.form.valid && projectSlug && modelSlug) {
+      const t4cInstanceId = this.form.value.t4cInstanceId!;
+      const t4cRepositoryName = this.form.value.t4cRepositoryName!;
+      const t4cProjectName = this.form.value.t4cProjectName!;
```

Comment:
Can't you use the getter from above here? (`get t4cProjectNameControl()` and `t4cRepositoryNameControl()`)

Reply:
I usually use `form.value.xy` instead of `form.controls.xy.value` to safe the additional `controls` but as I already have the two `get` for the control it is indeed better to use them 👍